### PR TITLE
fix(query): fix parsing of queries having aggregate function in metric name

### PR DESCRIPTION
**Pull request by TanviBhavsar**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** 
Parsing of queries with aggregate function in name  throws exception : java.lang.IllegalArgumentException: Aggregate functions have at least 1 parameter and utmost 2 parameters\n\tat 
Example: `sum:some_metric:dataset:1m{_ws_="some-workspace", _ns_="some-namespace"}`

**New behavior :**
Make first functionParams mandatory in aggregateExpression. Also added another aggregateExpression so that queries with aggregateGrouping before and after metric name are supported. 
Example: sum without (foo) (some_metric) and sum (some_metric) without (foo).

### CODE REVIEW

File: prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala [hunk 1]
```diff
@@ -187,7 +187,7 @@ class ParserSpec extends FunSpec with Matchers {
     parseSuccessfully("count_values(\"value\",some_metric)")
     parseSuccessfully("sum without(and, by, avg, count, alert, annotations)(some_metric)")
     parseSuccessfully("sum_over_time(foo)")
-
+    parseSuccessfully("sum:some_metric:dataset:1m{_ws_=\"demo\", _ns_=\"test\"}")
```
Comment (sherali42) [inline prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala]: @TanviBhavsar can you please add a test case to expected logicalPlans? `Should be able to make logical plans for Series Expressions`

Comment (whizkido) [reply prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala]: Also lets add all the keywords to these test cases , for completeness. And I agree with Sher's suggestion too.

Comment (TanviBhavsar) [reply prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala]: Added test case to expected logicalPlan. Also added test case for other keywords

**Comment by whizkido** [comment]
lgtm

**Pull request merged**